src/lib/builtins/command_info.rs

```diff
@@ -0,0 +1,85 @@
+use sys;
+use shell::Shell;
+use shell::status::*;
+use builtins::man_pages::*;
+
+use std::env;
+use std::path::Path;
+
+pub(crate) fn which(args: &[&str], shell: &mut Shell) -> Result<i32, String> {
+    if check_help(args, MAN_WHICH) {
+        return Ok(SUCCESS)
+    }
+
+    if args.len() == 1 {
+        return Err(String::from("which: Expected at least 1 args, got only 0"))
```

Comment:
May want to just write this directly to standard error, and change the error return value to `()` to avoid the allocation.